src/cloudant/design_document.py

```diff
@@ -149,6 +160,21 @@ def add_search_index(self, index_name, search_func, analyzer=None):
 
         self.indexes.__setitem__(index_name, search)
 
+    def add_list_function(self, list_name, list_func):
+        """
+        Appends a list function to the locally cached DesignDocument
+        indexes dictionary.
+
+        :param list_name: Name used to identify the list function.
+        :param list_func: Javascript list function.
```

Comment:
For friendlier rendering in the documentation the param list should look like:

```
param str list_name: ...
param str list_func: ...
```

You should apply that throughout your newly added code since you are leaving out the param "type" in a lot of your docstrings.